### PR TITLE
fix: unlock mutex for experiment ResourcePool() [RM-152]

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Ticket
RM-152

## Description

Unlock the internal experiment mutex in ResourcePool()

## Test Plan
I added a unit test that checks that the function returns smoothly. Without my change, the test would timeout anyway/fail.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[RM-152]: https://hpe-aiatscale.atlassian.net/browse/RM-152?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/experiment_job_service_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	rpName := e.ResourcePool()
+	require.Equal(t, rp, rpName)
+	require.Equal(t, mutexLocked(&e.mu), false)
+}
+
+func mutexLocked(m *sync.Mutex) bool {
+	state := reflect.ValueOf(m).Elem().FieldByName("state")
+	return state.Int()&1 == 1
+}
```
Comment (NicholasBlaskey) [inline master/internal/experiment_job_service_test.go:27]: I think by getting the ResourcePool you know the mutex is unlocked.

I think we should not go this deep on testing the mutex and just assuming if ResourcePool returns the right awsner the code is right

File: master/internal/experiment_job_service.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func (e *internalExperiment) SetResourcePool(resourcePool string) error {
 // ResourcePool gets the experiment's resource pool.
 func (e *internalExperiment) ResourcePool() string {
```
Comment (NicholasBlaskey) [inline master/internal/experiment_job_service.go:83]: release note?

**Review by NicholasBlaskey — APPROVED** [review]
one crazy idea is to try and right a regex that catches 

```defer *.Unlock()``` and put it in forbidigo

https://github.com/determined-ai/determined/blob/65339d21d2daaa62f1dcff9d8ffa2a3ff35e8d21/master/.golangci.yml#L77

File: master/internal/api_project_intg_test.go [hunk 3]
```diff
@@ -32,8 +32,10 @@ import (
 
 var pAuthZ *mocks.ProjectAuthZ
 
+const mockName = "mock"
```
Comment (carolinaecalderon) [inline master/internal/api_project_intg_test.go:35]: note: had to add these changes to satisfy the linter, although my PR doesn't touch this

File: master/internal/experiment_job_service.go (hunk 2 above)
Comment (carolinaecalderon) [reply master/internal/experiment_job_service.go:83]: added

File: docs/release-notes/experiment-deadlock.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**Bug Fixes**
+
+-  Kubernetes: Fix a bug where reading an experiment's resource pool name would hang indefinitely
```
Comment (NicholasBlaskey) [inline docs/release-notes/experiment-deadlock.rst:5]: I don't think it Kubernetes specific and I think it came on after move job so I would try and relate the release note to what customers would experience

Maybe something like

```
API: Fix an issue where calling ``det job update`` could cause jobs to be prevented from scheduling and ``det job ls`` to hang. 
```

**Review by NicholasBlaskey — APPROVED** [review]

File: docs/release-notes/experiment-deadlock.rst [hunk 5]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**Bug Fixes**
+
+-  API: Fix an issue where calling ``det job update`` could cause jobs to be prevented from
+   scheduling and ``det job ls`` to hang.
```
Comment (tara-det-ai) [inline docs/release-notes/experiment-deadlock.rst:6]: :orphan:

**Bug Fixes**

-  API: Fix a bug where calling ``det job update`` could prevent jobs from being scheduled and ``det job ls`` to hang.

**Review by tara-det-ai — APPROVED** [review]
suggested edit

**Pull request merged**